pandas/core/window/rolling.py

```diff
@@ -1124,14 +1034,17 @@ def _apply(
             whether to cache a numba compiled function. Only available for numba
             enabled methods (so far only apply)
         **kwargs
-            additional arguments for rolling function and window function
+            additional arguments for scipy windows if necessary
 
         Returns
         -------
         y : type of input
         """
-        win_type = self._get_win_type(kwargs)
-        window = self._get_window_weights(win_type=win_type)
+        signal = import_optional_dependency(
+            "scipy.signal", extra="Scipy is required to generate window weight."
+        )
+        assert self.win_type is not None  # for mypy
+        window = getattr(signal, self.win_type)(self.window, **kwargs)
```

Comment:
add a test if you pass bogus kwargs to a real scipy function.

Reply:
Done